packages/webdriverio/src/types.ts

```diff
@@ -70,7 +70,7 @@ export interface ChainablePromiseArray<T> extends Promise<T> {
     /**
      * allow to access a specific index of the element set
      */
-    [n: number]: WebdriverIO.Element | undefined
+    [n: number]: Promise<WebdriverIO.Element | undefined>
```

Comment:
This should be a `ChainablePromiseElement` because you can do things like: `$$('foobar')[1].click()`

```suggestion
    [n: number]: ChainablePromiseElement<WebdriverIO.Element | undefined>
```